t/17-basetest.t

```diff
@@ -22,6 +23,29 @@ use needle;
 sub write_with_thumbnail {
 }
 
+subtest run_post_fail_test => sub {
+    my $basetest_class = 'basetest';
+    my $autotest       = Test::MockModule->new("autotest");
+    $autotest->noop('set_current_test');
+    my $mock_basetest = Test::MockModule->new($basetest_class);
+    $mock_basetest->noop('take_screenshot');
+    $mock_basetest->mock(run => sub { die(); });
+    my $basetest = bless({
+            details      => [],
+            name         => 'foo',
+            category     => 'category1',
+            execute_time => 42,
+    }, $basetest_class);
+
+    dies_ok { $basetest->runtest } 'run_post_fail end up with die';
+
+    #behavior persists even after define _SKIP_POST_FAIL_HOOKS variable
+    $bmwqemu::vars{_SKIP_POST_FAIL_HOOKS} = 1;
+
+    dies_ok { $basetest->runtest } 'run_post_fail end up with die';
+
```

Comment:
Remove blank lines in between. Also to make the test strings unique and provide the text you have as in-line comment during runtime I suggest the following:

```suggestion
    $bmwqemu::vars{_SKIP_POST_FAIL_HOOKS} = 1;
    dies_ok { $basetest->runtest } 'run_post_fail behavior persists regardless of _SKIP_POST_FAIL_HOOKS setting';
```